### PR TITLE
Revert api changes

### PR DESCRIPTION
**Pull request by shireenf-ibm**
issue #236 

sub task:

> *  - [ ] revert changes to connlist API funcs, add new func to get the []ExposedPeer
>              Also, do not change types for existing APIs

### CODE REVIEW

**Comment by shireenf-ibm** [comment]
the only exposed function that I did not revert its changes :
`func (pe *PolicyEngine) AddPodByNameAndNamespace(name, ns string, nsLabels map[string]string) (Peer, error) `

it is exposed for `eval` pkg but used in internal function in `connlist` 
so I could not decide if it satisfies the requirement of a new major version!!

File: pkg/netpol/connlist/exposed_peer.go [hunk 1]
```diff
@@ -34,5 +32,5 @@ type XgressExposureData interface {
 	// PodLabels are matchLabels of potential pods which the peer might be exposed to
 	PodLabels() map[string]string
 	// PotentialConnectivity the potential connectivity of the exposure
-	PotentialConnectivity() conn.AllowedSet
+	PotentialConnectivity() common.Connection
```
Comment (adisos) [inline pkg/netpol/connlist/exposed_peer.go:35]: I'm not sure want the return value to be an interface type from internal package. 
Please add another sub-task to address this later.

File: pkg/netpol/connlist/connlist.go [hunk 2]
```diff
@@ -176,6 +165,11 @@ func (ca *ConnlistAnalyzer) Errors() []ConnlistError {
 	return ca.errors
 }
 
+// ExposureAnalysis returns a slice of ExposedPeer with all exposed pods and their exposure data
```
Comment (adisos) [inline pkg/netpol/connlist/connlist.go:168]:
```suggestion
// ExposureAnalysis returns a slice of ExposedPeer objects,  capturing for input workloads if/how they  may
// be exposed to potential cluster entities, based on the input network policies
```

File: pkg/netpol/connlist/connlist.go [hunk 3]
```diff
@@ -176,6 +165,11 @@ func (ca *ConnlistAnalyzer) Errors() []ConnlistError {
...
+func (ca *ConnlistAnalyzer) ExposureAnalysis() []ExposedPeer {
```
Comment (adisos) [inline pkg/netpol/connlist/connlist.go:169]: consider renaming function to `ExposedPeers()` instead

File: pkg/netpol/connlist/connlist.go [hunk 4]
```diff
@@ -195,27 +189,37 @@ func (ca *ConnlistAnalyzer) hasFatalError() error {
...
 	return ca.getConnectionsList(pe, ia)
 }
 
-// ConnlistFromK8sCluster returns the allowed connections list from k8s cluster resources, a list of all peers names,
-// and list of exposed peers in the parsed resources and their potentially allowed connections
-// if exposure-analysis option is on, otherwise nil
-func (ca *ConnlistAnalyzer) ConnlistFromK8sCluster(clientset *kubernetes.Clientset) ([]Peer2PeerConnection, []Peer, []ExposedPeer, error) {
-	pe := eval.NewPolicyEngine(ca.exposureAnalysis)
+// ConnlistFromK8sCluster returns the allowed connections list from k8s cluster resources, and list of all peers names
+// if exposure analysis option is on , performs exposure analysis computations
```
Comment (adisos) [inline pkg/netpol/connlist/connlist.go:214]:
```suggestion
```

File: pkg/netpol/eval/exposure.go [hunk 5]
```diff
@@ -26,8 +26,8 @@ import (
 
 // this file contains eval.PolicyEngine funcs which are related to exposure-analysis feature
 
-// addPotentiallyExposedPods adds pods which are potentially exposed to connect with (from/to) existing pods in the resources
-func (pe *PolicyEngine) addPotentiallyExposedPods() error {
+// addInferredFromPoliciesPods adds representative pods which are exposed to connect with (from/to) existing pods in the resources
+func (pe *PolicyEngine) addInferredFromPoliciesPods() error {
```
Comment (adisos) [inline pkg/netpol/eval/exposure.go:30]: rename func to `addRepresentativePods`

File: pkg/netpol/eval/exposure.go [hunk 6]
```diff
@@ -26,8 +26,8 @@ import (
 
 // this file contains eval.PolicyEngine funcs which are related to exposure-analysis feature
 
-// addPotentiallyExposedPods adds pods which are potentially exposed to connect with (from/to) existing pods in the resources
-func (pe *PolicyEngine) addPotentiallyExposedPods() error {
+// addInferredFromPoliciesPods adds representative pods which are exposed to connect with (from/to) existing pods in the resources
```
Comment (adisos) [inline pkg/netpol/eval/exposure.go:29]:
```suggestion
// addInferredFromPoliciesPods adds representative pods by inferring from network policies selectors within rules
// the required entities to represent (namespaces or pods by certain labels selectors)
// for example, if a rule within policy has namespace selector "name: foo", then a representative pod in such a 
// namespace with those labels will be added, representing all potential pods in such a namespace
```

**Review by adisos — CHANGES_REQUESTED** [review]
added a few comments.
you can keep the func's `AddPodByNameAndNamespace` new version as it is.

File: pkg/netpol/eval/exposure.go (hunk 5 above)
Comment (shireenf-ibm) [reply pkg/netpol/eval/exposure.go]: done

File: pkg/netpol/connlist/connlist.go (hunk 3 above)
Comment (shireenf-ibm) [reply pkg/netpol/connlist/connlist.go]: done

File: pkg/netpol/connlist/exposed_peer.go (hunk 1 above)
Comment (shireenf-ibm) [reply pkg/netpol/connlist/exposed_peer.go]: a solution I could think of is to expose common.Connection as connlist.AllowedSet ( so we don't duplicate the interface implementation again)  - done on [commit](https://github.com/np-guard/netpol-analyzer/pull/298/commits/bd98185794a50023ff2d74db9c1307007dd681aa)

Comment (adisos) [reply pkg/netpol/connlist/exposed_peer.go]: let's revert this commit and think about other options later.. should be done in a separate sub task

**Review by adisos — APPROVED** [review]

**Pull request merged**